e2e-tests/testing.md

```diff
@@ -0,0 +1,386 @@
+# End-to-end tests
+
+## Explanation
+
+The end-to-end tests are implemented using [YARF](https://github.com/canonical/yarf). They cover a wide range of scenarios,
+both for authd and the brokers.
+
+## Running the tests locally
+
+Running the tests locally requires a bit of setting up. This is a step-by-step guide to get you started.
+
+### 1. Install the required dependencies
+
+The tests have mainly two sets of dependencies: one required to configure and run the VM and one to build YARF from source.
+
+- Virtualization dependencies:
+
+    ```text
+    libvirt0 libvirt-clients libvirt-clients-qemu libvirt-daemon libvirt-daemon-system libvirt-daemon-driver-qemu qemu-system-x86 qemu-utils qemu-kvm socat sshpass wget cloud-image-utils
+    ```
+
+    Those are all part of the archive and can be installed on Ubuntu with:
+
+    ```bash
+    sudo apt-get install -y {virtualization dependencies}
+    ```
+
+- Test-run dependencies:
+
+    ```text
+    clang libxkbcommon-dev libcairo2-dev libgirepository-2.0-dev tesseract-ocr python3-tk python3-gi python3-cairo xvfb ffmpeg libwebkit2gtk-4.1-0 libwebkit2gtk-4.1-dev libgtk-3-0t64 libgtk-3-dev
+    ```
+
+    Those are all part of the archive and can be installed on Ubuntu with:
+
+    ```bash
+    sudo apt-get install -y {test-run dependencies}
+    ```
+
+### 2. Setup the VM
+
+The tests need a VM to run. This can be easily setup by using the domain definition and cloud-init configuration provided in the repository.
+
+1. Download the latest Ubuntu Desktop image (and resize it):
+
+    ```bash
+    wget https://cloud-images.ubuntu.com/questing/current/questing-server-cloudimg-amd64.img -O {img_path}
+
+    qemu-img resize {img_path} 10G
+    ```
+
+2. Create the cloud-init iso using the provided configuration in `e2e-tests/vm/runner-cloud-cfg.yaml`:
+   1. Make sure to update the YAML file and change the `E2E_VM_PASSWORD` placeholder with the desired password.
```

Comment:
can we just hardcode the password to make the setup steps simpler? I would use `ubuntu` as the hardcoded password.